scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/GoalBuilder.java

```diff
@@ -138,9 +144,49 @@ private static ActivityCreationTemplate makeActivityTemplate(
       }
     }
     builder = builder.ofType(type);
-    for (final var argument : activityTemplate.arguments().entrySet()) {
+    for (final var argument : canonicalizeSerialization(type, activityTemplate.arguments()).entrySet()) {
       builder = builder.withArgument(argument.getKey(), argument.getValue());
     }
     return builder.build();
   }
+
+  /**
+   * Take a set of arguments and run it through getEffectiveArguments to make sure that the SerializedValues
+   * are the correct type. E.g. Int -> Double.
```

Comment:
> `E.g. Int -> Double`

This is an example of `SerializedValues` that are not correct, right?

Reply:
Ohh it's an example of types that TS does not differentiate

Reply:
There are _ways_ to make TS differentiate, but its definitely not a simple thing and can mess with type usage generally (see https://www.typescriptlang.org/play#example/nominal-typing)

Reply:
We do it with our timestrings in the command expansion library here https://github.com/NASA-AMMOS/aerie/blob/develop/command-expansion-server/src/lib/codegen/CommandEDSLPreface.ts#L325-L326

Reply:
Thanks!!

Reply:
I _think_ we _could_ reliably do this with an int check to cast it to int internal to our type system, but again I'm not sure that's really a solution for this case as it doesn't get passed through to json